tools/2024-06-gitops-feature-spec.md

```diff
@@ -0,0 +1,374 @@
+
+## Desired customer experience outcome
+<!-- <Write this as an “I statement” that expresses the new capability from customer perspective … i.e. After this scenario is implemented “I can do, x, y, z, steps in cloud native app developer and seamlessly blah blah blah …. As a result <summarize positive impact on your work / business>  -->
+As an example, the application team will require 10 replica pods for their prod environment however this variable is currently set to 1 in the current [Bicep parameters](https://learn.microsoft.com/en-us/azure/azure-resource-manager/bicep/parameter-files?tabs=Bicep) file created by the operator or developer called `app.bicepparams` in the parameters git repo:
+
+````bicep
+using './app.bicep'
+
+param replicaCount = 1
+````
+
+The above parameter is defined and used in the `app.bicep` file stored in a separate repo as follows:
+
+```bicep
+import radius as radius
+
+@description('The app ID of your Radius Application. Set automatically by the rad CLI.')
+param application string
+param replicaCount int
+
+resource demo 'Applications.Core/containers@2023-10-01-preview' = {
+  name: 'demo'
+  properties: {
+    application: application
+    container: {
+      image: 'ghcr.io/radius-project/samples/demo:latest'
+      ports: {
+        web: {
+          containerPort: 3000
+        }
+      }
+      extensions: [
+      {
+        kind:  'manualScaling'
+        replicas: replicaCount
+      }
+    ]
+    }
+  }
+}
+```
+
+To fix this as a SRE, I can:
+
+1. Open the `app.bicepparams` file in my editor of choice and change the `replicaCount` parameter to 10.
+
+````bicep
+using './app.bicep'
+
+param replicaCount = 10
+````
+
+2. Commit and push the change to the parameters git repository.
+
+2. Flux/ArgoCD detects the change and triggers Radius to redeploy the `demo` container resources with the new replica count. Radius is able to read the `app.bicepparams` file and apply the new replica count to the `demo` container resources, despite the `app.bicep` file being unmodified and also stored in a separate git repository.
+
+> As an SRE, I'm very happy with this experience because I did not have to do many steps to apply my parameter patch and while I had to learn some Bicep knowledge and Radius knowledge, I was able to ultimately leverage Flux or ArgoCD mechanisms which I'm familiar with to apply my patches to the Kubernetes cluster.
```

Comment:
What if the configuration that the user wants to edit is not exposed by our API? e.g. `ImagePullPolicy` or something

Reply:
podspec patching

Reply:
added a note for this.